js/app/dashboard/components/TaskList.js

```diff
@@ -137,199 +115,163 @@ const ProgressBarMemo = React.memo(({
     )
   })
 
-class TaskList extends React.Component {
+export const TaskList = ({ uri, username, distance, duration }) => {
+  const dispatch = useDispatch()
+  const unassignTasksFromTaskList = (username => tasks => dispatch(unassignTasks(username, tasks)))(username)
 
-  remove(task) {
-    this.props.unassignTasks(this.props.username, task)
-  }
+  const selectTaskListItems = makeSelectTaskListItemsByUsername()
```

Comment:
I don't really understand why we need to wrap `createSelector` inside a function in `makeSelectTaskListItemsByUsername`. What does not work if we make it a usual selector?

Reply:
because of this : https://redux.js.org/usage/deriving-data-selectors#createselector-behavior

actually if you pass ID directly to the selector you never get the cached result you re-run the selector each time because the selector cache size is one and now you are changing arguments

so it is better to just create a new selector for each different ID/component

it is how far i am understanding, tell me if it is not clear